org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/regionaccess/internal/RegionAccessBuilderTest.java

```diff
@@ -0,0 +1,805 @@
+				"21 1  H \" \"        Whitespace:TerminalRule'WS'\n" +
+				"22 5  S \"child\"    Fragment:'child'\n" +
+				"27 1  H \" \"        Whitespace:TerminalRule'WS'\n" +
+				"      B Action     Fragment:mixed=Mixed path:Action/mixed\n" +
+				"28 1   S \"(\"        Mixed:'('\n" +
+				"29 0   H\n" +
+				"29 1   S \")\"        Mixed:')'\n" +
+				"      E Action     Fragment:mixed=Mixed path:Action/mixed\n" +
+				"30 0  H\n" +
+				"30 1  S \")\"        Mixed:')'\n" +
+				"     E Action     Root\n" +
+				"31 0 H\n";
+		operator_tripleEquals(model,
+				expectation);
+	}
+
+	@Test
+	public void testMixedRootAction() throws Exception {
+		String model = "7 action (foo)\n".trim();
+		String expectation =
+				" 0 0 H\n" +
+				"     B RootAction Root\n" +
+				" 0 1  S \"7\"        Root:'7'\n" +
+				" 1 1  H \" \"        Whitespace:TerminalRule'WS'\n" +
+				" 2 6  S \"action\"   Root:'action'\n" +
+				" 8 1  H \" \"        Whitespace:TerminalRule'WS'\n" +
+				"      B Mixed'foo' Root:mixed=Mixed path:RootAction/mixed\n" +
+				" 9 1   S \"(\"        Mixed:'('\n" +
+				"10 0   H\n" +
+				"10 3   S \"foo\"      Mixed:name=ID\n" +
+				"13 0   H\n" +
+				"13 1   S \")\"        Mixed:')'\n" +
+				"      E Mixed'foo' Root:mixed=Mixed path:RootAction/mixed\n" +
+				"     E RootAction Root\n" +
+				"14 0 H\n";
+		operator_tripleEquals(model,
+				expectation);
+	}
+
+	@Test
+	public void testComments1() throws Exception {
+		String model =
+				"/*xxxx*/\n" +
+				"8\n" +
+				"/*aaaaa*/\n" +
+				"c\n" +
+				"// last\n".trim();
+		String expectation =
+				" 0    H \"/*xxxx*/\" Comment:TerminalRule'ML_COMMENT' Association:NEXT\n" +
+				"    9   \"\\n" +
+				"\"       Whitespace:TerminalRule'WS'\n" +
+				"      B ValueList'[c]' Root\n" +
+				" 9  1  S \"8\"        Root:'8'\n" +
+				"10     H \"\\n" +
+				"\"       Whitespace:TerminalRule'WS'\n" +
+				"         \"/*aaa...\" Comment:TerminalRule'ML_COMMENT' Association:NEXT\n" +
+				"   11    \"\\n" +
+				"\"       Whitespace:TerminalRule'WS'\n" +
+				"21  1  S \"c\"        ValueList:name+=ID\n" +
+				"      E ValueList'[c]' Root\n" +
+				"22    H \"\\n" +
+				"\"       Whitespace:TerminalRule'WS'\n" +
+				"    8   \"// last\"  Comment:TerminalRule'SL_COMMENT' Association:CONTAINER\n";
+		operator_tripleEquals(model,
+				expectation);
+	}
+
+	private void operator_tripleEquals(CharSequence file, CharSequence expectation) throws Exception {
+		String exp = expectation.toString();
+		Root obj = parseHelper.parse(file);
+		validationTestHelper.assertNoErrors(obj);
+		ITextRegionAccess access1 = createFromNodeModel(obj);
+		ITextRegionAccess access2 = serializer.serializeToRegions(obj);
+		assertToStringDoesNotCrash(access1);
+		assertToStringDoesNotCrash(access2);
+		assertLinesAreConsistent(access1);
+		assertLinesAreConsistent(access2);
+		TextRegionAccessToString _cfg = cfg(new TextRegionAccessToString().withRegionAccess(access1));
```

Comment:
unnecessary local vars ... I wouldn't mind but the underscore prefix is a little too alien for my taste.